superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -370,7 +370,7 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
             className="preferred-item"
             onClick={() => setDatabaseModel(database.engine)}
             buttonText={database.name}
-            icon={dbImages[database.engine]}
+            icon={dbImages && dbImages[database.engine]}
```

Comment:
I think this might be better as a const in the component that you then call. 

You might also want to check this with no image because I am pretty sure icon is a required class on this.

Reply:
I want to keep the component functional, and i add a condition for the icon inside the compenent. Basically this will default to the default icon if icon isnt set